### PR TITLE
API-10485: Refactor Reports API

### PR DESCRIPTION
**Pull request by skamieniarz**

### CODE REVIEW

File: livechat/reports/base.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            Args:
+                token (str): Full token with type (Bearer/Basic) that will be
+                             used as `Authorization` header in requests to API.
+                version (str): API's version. Defaults to the stable version of API.
+                base_url (str): API's base url. Defaults to API's production URL.
+                http2 (bool): A boolean indicating if HTTP/2 support should be
+                              enabled. Defaults to `False`.
+
+            Returns:
+                ReportsApiV: API client object for specified version.
```
Comment (skamieniarz) [inline livechat/reports/base.py:40]: `V` to be removed.

File: livechat/reports/base.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+''' Base class that allows retrieval of client for specific Reports
```
Comment (kacperf531) [inline livechat/reports/base.py:1]: Base module/Module with base class*

File: livechat/reports/api/v33.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
+''' Reports API client class in version 3.3. '''
```
Comment (kacperf531) [inline livechat/reports/api/v33.py:1]: These also should be changed to hint they're module docstrings, e.g.
```suggestion
''' Reports API client module for v3.3. '''
```

**Review by kacperf531 — DISMISSED** [review]

File: livechat/reports/base.py (hunk 2 above)
Comment (skamieniarz) [reply livechat/reports/base.py]: Fixed in 3c5b25632a401cd9fc34a0137405e70c83b297a9.

File: livechat/reports/api/v33.py (hunk 3 above)
Comment (skamieniarz) [reply livechat/reports/api/v33.py]: Fixed in 3c5b25632a401cd9fc34a0137405e70c83b297a9.

**Review by kacperf531 — APPROVED** [review]

**Pull request merged**